dataprofiler/tests/profilers/test_column_profile_compilers.py

```diff
@@ -680,6 +609,165 @@ def test_json_decode_after_update(self):
+
+        # Test disabling both datalabeler profiles for compiler diff
+        compiler2 = col_pro_compilers.ColumnDataLabelerCompiler(data, options)
+        expected_diff = {}
+        self.assertDictEqual(expected_diff, compiler1.diff(compiler2))
+
+    def test_json_encode(self):
+
+        compiler = col_pro_compilers.ColumnDataLabelerCompiler()
+
+        serialized = json.dumps(compiler, cls=ProfileEncoder)
+        expected = json.dumps(
+            {
+                "class": "ColumnDataLabelerCompiler",
+                "data": {
+                    "name": None,
+                    "_profiles": {},
+                },
+            }
+        )
+        self.assertEqual(expected, serialized)
+
+    def test_json_encode_after_update(self):
+
+        data = pd.Series(["-2", "-1", "1", "2"], name="test")
+        with test_utils.mock_timeit():
+            compiler = col_pro_compilers.ColumnDataLabelerCompiler(data)
+
+        with mock.patch.object(
+            compiler._profiles["order"], "__dict__", {"an": "order"}
+        ):
+            with mock.patch.object(
+                compiler._profiles["category"], "__dict__", {"this": "category"}
+            ):
+                serialized = json.dumps(compiler, cls=ProfileEncoder)
+
+        expected = json.dumps(
+            {
+                "class": "ColumnDataLabelerCompiler",
+                "data": {
+                    "name": "test",
+                    "_profiles": {
+                        "order": {"class": "OrderColumn", "data": {"an": "order"}},
+                        "category": {
+                            "class": "CategoricalColumn",
+                            "data": {"this": "category"},
+                        },
+                    },
+                },
+            }
+        )
+
+        self.assertEqual(expected, serialized)
+
+    def test_json_decode(self):
+        expected_compiler = col_pro_compilers.ColumnDataLabelerCompiler()
+        serialized = json.dumps(expected_compiler, cls=ProfileEncoder)
+
+        deserialized = load_compiler(json.loads(serialized))
+
+        test_utils.assert_profiles_equal(expected_compiler, deserialized)
+
+    def test_json_decode_after_update(self):
+
+        data = pd.Series(["-2", "-1", "1", "15"], name="test")
+        with test_utils.mock_timeit():
+            expected_compiler = col_pro_compilers.ColumnDataLabelerCompiler(data)
+
+        serialized = json.dumps(expected_compiler, cls=ProfileEncoder)
+        deserialized = load_compiler(json.loads(serialized))
+
+        test_utils.assert_profiles_equal(deserialized, expected_compiler)
+        # assert before update
+        assert deserialized.report().get("order", None) == "ascending"
+        assert deserialized.report().get("categorical", None) == True
+
+        df_float = pd.Series(
+            list(range(100))  # make orer random and not categorical
+        ).apply(str)
+
```

Comment:
maybe a data case more specific to what we want to check in data labeling

Reply:
working this now